### DISCUSSION TITLE
Sanny Builder Library: Command Decision Tree

### DISCUSSION THREAD

**Discussion by x87** [sannybuilder/dev#162] [Showroom]
Hi all,

I added a new feature to Sanny Builder Library that I think is worth a new discussion. This is a  questionnaire or, more formally, a decision tree, that should help a novice scripter find a command suiting their needs.

Decision Trees can be found for each game under /find route:
https://library.sannybuilder.com/#/gta3/find
https://library.sannybuilder.com/#/vc/find
https://library.sannybuilder.com/#/sa/find

The tree is constructed out of predefined statements (or sentences). Each sentence describes only one command. 

Example of a statement:

```
"009A": "I want to create | a character   | standing on foot",
```

Vertical bars denote a choice that the user must made. Depending on other statements you may have one or more possible choices to narrow the search. Picking the given choices, you complete the statement and get the final command.

The tree is game AND language specific. The reason for that is, a game may have unique commands that would require its own statements. It is language specific as each language has its own grammar or punctuation rules. It's hard to split statements in steps and make them correctly display in all languages.

Shared statements are also available, they are included in any game's decision tree. 
https://github.com/sannybuilder/library/tree/master/shared/statements

Game specific statements live in the game folders:
https://github.com/sannybuilder/library/tree/master/gta3/statements
https://github.com/sannybuilder/library/tree/master/vc/statements
https://github.com/sannybuilder/library/tree/master/sa/statements

--

I only made a handful of statements as an example. I need community's help to get more statements in different languages. My initial goal is to describe at least 100-200 most used commands. 

Any feedback is very welcome 👍

**Comment by Vladik01-11** [comment]
I suggest adding the following for the Ru language:
  "03E5": "Я хочу вывести    | текст      | , в виде текст-бокса",
  "00BA": "Я хочу вывести    | текст      | , с каким-то стилем",
  "032B": "Я хочу создать    | пикап      | , с каким-то оружием | дальнего боя",
  "0213": "Я хочу создать    | пикап      | , с каким-то оружием | ближнего боя",
  "0213": "Я хочу создать    | пикап      | , с каким-то оружием | ближнего боя",
  "02E1": "Я хочу создать    | пикап      | , как при убийстве прохожего",
  "04A5": "Я хочу получить    | текущие координаты трупа",
  "09E1": "Я хочу узнать    | текущую цену автомобиля",
  "08B1": "Я хочу вкл/выкл | режим ночного виденья",
  "08B2": "Я хочу вкл/выкл | режим теплового виденья",
  "08EA": "Я хочу вкл/выкл | генератор создания членов банд",
  "08AC": "Я хочу вкл/выкл | зоны банд"
